code/go/0chain.net/blobbercore/challenge/entity.go

```diff
@@ -107,6 +111,11 @@ func unMarshalField(stringObj datatypes.JSON, dest interface{}) error {
 }
 
 func (cr *ChallengeEntity) Save(ctx context.Context) error {
+	db := datastore.GetStore().GetTransaction(ctx)
+	return cr.SaveWith(db)
+}
+
+func (cr *ChallengeEntity) SaveWith(db *gorm.DB) error {
```

Comment:
Seems the db is the only option for SaveWith, if so we should still use it the previous way to keep it simple

Reply:
NO. they are different.  it is a db session with heavy transaction if it is got from `GetTransaction(ctx)`. It is a normal db session without transaction in `SaveWith`.  that is why It is added. but `Save` is still used on other code with built-in transaction. it is not easy to refactor it. 
On blobber, transaction always be started in `WithConnection` and `WithReadonlyConnection` in each request. it is not good .but it is really difficult to refactor all of them.